### DISCUSSION TITLE
HmIP-BSM are flaged as light

### DISCUSSION THREAD

**Discussion by itn3rd77** [SukramJ/aiohomematic#84] [General]
Hi Markus,

the device channels :4, :5 and :6 of the HmIP-BSM are flaged as `light` but I think this should be `switch` as this is a generic  switch actor. Or am I wrong?

![grafik](https://user-images.githubusercontent.com/2655169/147114932-f277f5b3-a953-4f53-b954-2f60965b9af2.png)

**Comment by SukramJ** [comment]
I think light is correct. Most people use a BSM to control lights.
Same is with BDT. 
Basically light and switch are identical

**Reply by emufan** [reply]
Only to have the information here as well for the sake of completes: They are not the same. Switches cannot be grouped in light groups, which is a helpful platform for lights for UI and controls.

**Comment by itn3rd77** [comment]
Ok thanks. If there is no difference than it`s just a matter of taste.

**Comment by SukramJ** [comment]
You're right. I know it has been different with pyhomematic, but i already implemented it for homematicip_cloud this way.
For me it feels better as light.

**Reply by itn3rd77** [reply]
Hi Markus,

after checking my automations after transition to the new integration I have the feeling to come back to you regarding the switch/light discussion.

If using those devices in automations and you want to turn more than one device on or off you have to use choose `Light: On`, `Light: Off` or `Light: Toggle` and are faced with a bunch of options that are irrelevant, misleading and not usable for a device that is simply a switch. So I think it is not any longer matter of taste, or what do you think?

![grafik](https://user-images.githubusercontent.com/2655169/147241060-383d1aa8-17d8-434f-ac4a-fd8585dcc31d.png)

Best regards
Ingo

**Reply by SukramJ** [reply]
Yes, it is misleading in your case. Do you use a blueprint?

**Reply by itn3rd77** [reply]
No just the automation editor from inside Home Assistant

**Comment by SukramJ** [comment]
Why do you use a service instead of the device?

**Comment by itn3rd77** [comment]
For a single device that would be OK. But if you would  like to switch multiple devices or an area you have to choose the service.

**Reply by SukramJ** [reply]
What about service homeassitant.turn_on?

**Reply by itn3rd77** [reply]
Hehe ... ok that works and also hides the misusage as a light. But hey it's ok. I am fine without proselytize you.

**Reply by SukramJ** [reply]
It's not that i'am not thinking about to switch it.
Though I think the light is right, BSM is the only light, that is not a dimmer.

**Comment by emufan** [comment]
If someone else will come around this discussion in the future or more will participate:
I would vote for reverting this change and second every argument from SukramJ above. 

```
Version 0.3.1 (2021-12-23)
- Make HmIP-BSM a switch (only dimable devices should be lights)
```

but would of course accept if I'm the only one, who would like this and the maturity would like to have it as switch.

My reasons:
- my gut feeling is, that 95% will use the BSM as light (as SukramJ explained above as well)
- Such light should by default behave as lights in HA in terms of icons, classes, services, ...
- Main advantage 1: Only light can be included in light groups (which have a lot of advantages for lights in HA compared to simple groups). You can include lights in groups, but not switches in light groups.
- Main advantage 2: If using a light switch to get around the topic above, this is then decoupled from the device and integration, and you lose the new benefit of having a device, device view, device services and still have to use entities as before
- Migrating from HACloud integration will have the same and there

**Reply by SukramJ** [reply]
MY arguments for making it a switch:
- Only dimmers should be lights.
- Extra code is required for a single device, because it would be the only on/off light.
- There are also DIN-Rail-Switches that might be mostly used as light, how to deal with these devices.
- There are solutions in HA to make lights out of switches, if needed. I personally only changed the icon.

**Reply by emufan** [reply]
> * Only dimmers should be lights.

Every dimmer might be a ligt, but not every light is a dimmer. The obove is a rule, but should not be a reason. But most proably it is only because of:

> * Extra code is required for a single device, because it would be the only on/off light.

Yes. That's true. Be as you had this before 0.3.1 and as your own arguments in HMCloud as well, is should be a  common siolution.

> * There are also DIN-Rail-Switches that might be mostly used as light, how to deal with these devices.

You are right. But there I do not have a gut feeling, how many users use this as light or switch compared to my gut feeling for BSM

> * There are solutions in HA to make lights out of switches, if needed. I personally only changed the icon.

Yes. But with bought in disadvantages as described above. Decoupling from integration, lose of device connection in device view and generation, ...

I think both positions, the one from itn3rd77 / the new from you and the one from me / the old from you might be better. As I said. If I stay the only one, fine. Let's see. ☑

**Reply by itn3rd77** [reply]
As we are talking about an integration for Home Assistant and thus the single point of truth from my point of view is how Home Assistant defines [Lights](https://www.home-assistant.io/integrations/light/) and [Switches](https://www.home-assistant.io/integrations/switch/).

Taken this in to account a HmIP-BSM is a switch (period). HmIP-BSM doen't support a **_single_** service attribute listed for lights. You can't set transition, color brightness etc. for a device which is ... a simple switch. I have many HmIP-BSM devices. Some are used to switch the light on and some are used e.g. to switch an indor fountain on. You are mixing up your use case with the purpose of the device. Technical a HmIP-BSM is identical to a HMIP-PSM. If I plug in a floor lamp to an HMIP-PSM it is still a switch and not a light.

**Reply by emufan** [reply]
I accept, what you prefer. And accept the majority.

But your explanation is not completely correct. Please compare to the HUE and Osram Plugs via HUE integration. They are listed as light. And a https://www.home-assistant.io/integrations/light.switch/ is the same. And you have no real disadvantage to use a light instead of a switch, but have it (see above) when you have a switch instead of a light. I have explained the things around light groups, https://www.home-assistant.io/integrations/light.switch/, etc. which are need to be decoupled created locally for each in this case and to repeat this, will not bring more infos.

**Reply by danielperna84** [reply]
My 2 cents: the devs at HA tend to be very _kleinlich_ about many things. And for the most part for - in the long run - good reasons. I do see the the benefit of providing this device as a `light`. But I don't think it's the correct thing to do, as technically, which we all know by now, it's only a `switch`. I have a HmIP-DRSI4 and DRSI1 (which is _typical_ for substituting an Eltako _Treppenhauslicht_), all used to control lights. Yet I wouldn't expect them to be `light`s. It's sold as a _Schaltaktor_, and only a subset of users will use it as a light. In case of the BSM the subset might actually be the majority, but nevertheless, it's a `switch`. And since Home Assistant is prepared for such scenarios via template lights etc., there already is a workaround in place.

That being said, Homematic isn't the only integration which provides switches that are used as lights. And - returning to the statement of being _kleinlich_ - maybe one day HA introduces a simple checkbox for exactly this purpose: turning `switch`es into `light`s straight from HA. At that point people could start _complaining_ "why isn't this a `switch`?", which then might lead to changing this back once again, possibly causing a breaking change. Which wouldn't have been necessary if this the device is accepted for what it is, a `switch`.

TLDR: In my opinion, if possible, we should avoid _Extrawürste_ if: there's at least a (common) manual workaround available, or a HA-native workaround is either planned or at least a possibility for the future.

**Reply by chrste90** [reply]
I just read the release notes for the upcoming release 2022.4 and it seems that there will be the HA solution you described:
https://rc.home-assistant.io/blog/2022/03/30/release-20224/#switch-as-x

**Reply by emufan** [reply]
> The newly created entity will appear in your original device.

First I thought: Great, great, great! Then I read and think: Hm. Let's see. I read it that way, that it is only a UI for the https://www.home-assistant.io/integrations/light.switch/ 

Then I still have 2 entities as with light-Switch and it seems that it is only working with devices.

**Reply by chrste90** [reply]
Oh right, but if you have two entities, maybe the new option to hide entities can help here.

**Reply by emufan** [reply]
Yes, this is (for me) the more helpful new option. 👍 The other one is more or less the light.switch.
